ci-automation/sdk_bootstrap.sh

```diff
@@ -84,7 +84,14 @@ function sdk_bootstrap() {
     copy_to_buildcache "sdk/${ARCH}/${vernum}" "${dest_tarball}"*
     cd -
 
-    # Create new tag in scripts repo w/ updated versionfile + submodules
-    update_and_push_version "sdk-${git_vernum}"
+    # Create new tag in scripts repo w/ updated versionfile + submodules.
+    # Also push the changes to the branch ONLY IF we're doing a nightly
+    #   build of the 'main' branch AND we're definitely ON the main branch.
+    local push_branch="false"
+    if   [[ "${version}" =~ ^main-[0-9.]+-nightly-[-0-9]+$ ]] \
```

Comment:
Two more things to check are the `coreos-overlay` and `portage-stable` references.
Currently the doc string in Jenkins tells this: `use a suffix for dev builds (e.g. alpha-3234.0.0-nightly-1234).` and by checking for the two references we can at least prevent to have something else than main pop up here. People may also retrigger the nightly build to test a `jenkins-os` change but that's hard to catch here.

Reply:
I'd like to suggest a different approach. Placeholder value for SDK_VERSION (sdk-main-nightly) that does the lookup and finds the last built nightly SDK (from version.txt somewhere).

But I also don't understand what is failing here: the os->manifest->sdk chain uses the last released SDK as a base and works correctly for nightlies, so why does this chain need special handling?

Reply:
> Two more things to check are the coreos-overlay and portage-stable references.
> Currently the doc string in Jenkins tells this: use a suffix for dev builds (e.g. alpha-3234.0.0-nightly-1234).

That's bad documentation (and a good catch), filed a PR to the CI to update the docstring. Will add a check for the submodules branches.

> People may also retrigger the nightly build to test a jenkins-os change but that's hard to catch here.

I don't think that's a pattern we should encourage - if for some reason someone needs to test the CI then starting the sdk job is way leaner.

Reply:
Updated.

Reply:
A placeholder plus a way of automatically updating the submodules would work, too - both seem to have pros and cons…

Reply:
> I'd like to suggest a different approach. Placeholder value for SDK_VERSION (sdk-main-nightly) that does the lookup and finds the last built nightly SDK (from version.txt somewhere).

This is the SDK bootstrap job, it generates a new SDK version (SDK `version` number is an input to that job). It then updates `version.txt` and generates a commit, so follow-up jobs use the correct (new build) SDK version.  

> But I also don't understand what is failing here: the os->manifest->sdk chain uses the last released SDK as a base and works correctly for nightlies, so why does this chain need special handling?

Normally, SDK builds push build changes to a tag but do not update the branch. That tag is the used downstream to continue the build pipeline with the right version. This section introduces special handling for nightly builds of the main branch, which will update the main branch of the `scripts` repo with the new versionfile. Motivation is to make development branches based on `main` use the correct SDK version and remove the necessity for hacks like e.g. this one: https://github.com/flatcar-linux/scripts/pull/235/files#diff-48b3a88e2d986f5123acefc4dd59b5e6ae86f45975cc28e105e3ed1b3815f420L4